webapp/src/components/ActivateAccountActionHandler.vue

```diff
@@ -1,14 +1,125 @@
+<template>
+  <div>
+    <div class="text-center" v-if="captcha_required && !success">
+      <v-container class="pa-0">
+        <v-row dense align="center" class="text-center">
+          <v-col cols="12" sm="">
+            <v-text-field
+                    v-model="payload.captcha.solution"
+                    label="Type CAPTCHA text here"
+                    prepend-icon="mdi-account-check"
+                    outline
+                    required
+                    :disabled="working"
+                    :rules="captcha_rules"
+                    :error-messages="captcha_errors"
+                    @change="captcha_errors=[]"
+                    @keypress="captcha_errors=[]"
+                    class="uppercase"
+                    ref="captchaField"
+                    tabindex="3"
+                    :hint="captcha_kind === 'image' ? 'Can\'t see? Hear an audio CAPTCHA instead.' : 'Trouble hearing? Switch to an image CAPTCHA.'"
+            />
+          </v-col>
+          <v-col cols="12" sm="auto">
+            <v-progress-circular
+                  indeterminate
+                  v-if="captchaWorking"
+            ></v-progress-circular>
+            <img
+                  v-if="captcha && !captchaWorking && captcha_kind === 'image'"
+                  :src="'data:image/png;base64,'+captcha.challenge"
+                  alt="Passwords can also be reset by sending an email to our support."
+            />
+            <audio controls
+                  v-if="captcha && !captchaWorking && captcha_kind === 'audio'"
+            >
+              <source :src="'data:audio/wav;base64,'+captcha.challenge" type="audio/wav"/>
+            </audio>
+            <br/>
+            <v-btn-toggle>
+              <v-btn text outlined @click="getCaptcha(true)" :disabled="captchaWorking"><v-icon>mdi-refresh</v-icon></v-btn>
+            </v-btn-toggle>
+            &nbsp;
+            <v-btn-toggle v-model="captcha_kind">
+              <v-btn text outlined value="image" aria-label="Switch to Image CAPTCHA" :disabled="captchaWorking"><v-icon>mdi-eye</v-icon></v-btn>
+              <v-btn text outlined value="audio" aria-label="Switch to Audio CAPTCHA" :disabled="captchaWorking"><v-icon>mdi-ear-hearing</v-icon></v-btn>
+            </v-btn-toggle>
+          </v-col>
```

Comment:
man we really need to encapsulate this. I think its now the third time in the code base

Reply:
that shouldnt be part of this PR. Just sayin'